static/js/Misc.jsx

```diff
@@ -1103,35 +1104,80 @@ function useHiddenButtons() {
     return [hideButtons, handleMouseOverAdminButtons];
 }
 
-const CategoryHeader =  ({children, type, data = [], edit = true,
-                            add_subcategory = true, reorder = false,
-                            add_source = false}) => {
+const CategoryHeader =  ({children, type, data = [], buttonsToDisplay = ["subcategory", "edit"]}) => {
   /*
   Provides an interface for using admin tools.
-  `type` is 'sources', 'books' or 'topics'
-  `data` is list when `type` === 'books' which tells us where we are in the TOC tree,
+  `type` is 'sources', 'cats', 'books' or 'topics'
+  `data` is list when `type` === 'cats' which tells us where we are in the TOC tree,
+        for `type` === 'books' it's the name of the book
         for `type` === 'topics' it's a dictionary of the topic object
         for `type` === 'sources' it's a list where the first item is topic slug and second item is source data
+  `buttonsToDisplay` is a list that says in the specified order we want all of the buttons in buttonOptions
    */
   const [editCategory, toggleEditCategory] = useEditToggle();
   const [addCategory, toggleAddCategory] = useEditToggle();
   const [reorderCategory, toggleReorderCategory] = useEditToggle();
   const [addSource, toggleAddSource] = useEditToggle();
+  const [addSection, toggleAddSection] = useEditToggle();
   const [hiddenButtons, setHiddenButtons] = useHiddenButtons(true);
-
-  const adminClasses = classNames({adminButtons: 1, hiddenButtons});
-  let adminButtonsSpan = null;
-  if (Sefaria.is_moderator && editCategory) {
-    if (data.length === 0) {  // at /texts or /topics
-      const url = type === 'topics' ? `/api/topic/reorder` : `/api/category?reorder=1`;
-      const redirect = type === 'topics' ? '/topics' : '/texts';
-      const origItems = type === 'topics' ? Sefaria.topic_toc : Sefaria.toc;
-      adminButtonsSpan = <ReorderEditor close={toggleEditCategory} type={type} origItems={origItems}
-                                        postURL={url} redirect={redirect}/>;
-    } else if (type === "sources") {
-      const [topicSlug, refData] = data;
-      adminButtonsSpan = <SourceEditor topic={topicSlug} origData={refData} close={toggleEditCategory}/>;
-    } else if (type === "books") {
+  const buttonOptions = {"subcategory": ["Add sub-category", toggleAddCategory],
+                          "source": ["Add a source", toggleAddSource],
+                          "section": ["Add section", toggleAddSection],
+                          "reorder": ["Reorder sources", toggleReorderCategory],
+                          "edit": ["Edit", toggleEditCategory]};     
+
+  const getEditorForReordering = (sources) => {
```

Comment:
As more of the code is cleared it become easier to be nitpicky :) 

Here is an example of how you can modulate this code to get a cleaner pattern in which the components and url creation are done separately 

```
const createURLs = (type, data) => {
  if (type === 'topics') {
    return {
      url: '/api/topic/reorder',
      redirect: '/topics',
      origItems: Sefaria.topic_toc
    };
  }

  if (type === 'sources') {
    return {
      url: `/api/source/reorder?topic=${data.slug}&lang=${Sefaria.interfaceLang}`,
      redirect: `/topics/${data.slug}`,
      origItems: data.refs?.about?.refs || [],
    };
  }

  return {
    url: '/api/category?reorder=1',
    redirect: '/texts',
    origItems: Sefaria.toc
  };
};

const filterAndSortRefs = (refs) => {
  const filteredRefs = refs.filter((x) => !x.is_sheet && x?.order?.availableLangs?.includes(Sefaria.interfaceLang.slice(0, 2)));
  return filteredRefs.sort((a, b) => refSort('relevance', [a.ref, a], [b.ref, b])).slice(0, 30);
}

const getEditorForReordering = (sources) => {
  const { url, redirect, origItems } = createURLs(type, data);
  
  const refs = type === 'sources' ? filterAndSortRefs(origItems) : origItems;

  const closeAction = type === 'sources' ? toggleReorderCategory : toggleEditCategory;

  return (
    <ReorderEditor 
      close={closeAction} 
      type={type} 
      origItems={refs} 
      postURL={url} 
      redirect={redirect}
    />
  );
};

```